### ISSUE TITLE
Running neuron#edit_zettel_select before opening a zettel fails

### ISSUE THREAD

**Issue by MaienM** [fiatjaf/neuron.vim#2]
As per the title, running `:call neuron#edit_zettel_select()` before opening a zettel fails because the cache is not yet populated. The same likely applies to other methods that use the cache.

```
Error detected while processing function neuron#edit_zettel_select:                                                                                    
line    1:
E121: Undefined variable: g:list_pair_zettelid_zetteltitle
E116: Invalid arguments for function fzf#wrap
E116: Invalid arguments for function fzf#run
```

If I manually run `:call neuron#refresh_cache()` and then `:call neuron#edit_zettel_select()` it does works, confirming the problem is the cache.

I'd expect that any method that needs data from the cache would first populate this cache if this hasn't happened yet.

**Comment by fiatjaf** [comment]
What should the plugin do if the cache is being populated in the background already but you were too fast and called the function before it finished? Start a process to populate it again synchronously? My impression is that that would be worse for the user.

Or perhaps there is a way to pause a command until the cache is populated?
Or perhaps show a message "cache being populated, try this command again in one second"?

**Comment by MaienM** [comment]
I think that if the cache was not being populated yet this process should be started. If it's already running just printing a message "cache being populated, please wait" seems fine. It might be nice to show a message once it's done as well if you've previously gotten a "please wait" message.

Remembering what the action the user tried to perform was and triggering it once the cache is loaded would be nice, but low-priority as far as I'm concerned. Might also be more trouble than it's worth.

**Comment by fiatjaf** [comment]
Wait, what do you mean by "before opening a zettel"? You mean just starting vim with a blank unnamed buffer?

How should I proceed in this case? The plugin only runs when the file is "detected" to be a relevant file, right? I'm not sure when in the code is this happening, but it doesn't work with unnamed buffers. Apparently `plugin/neuron.vim` doesn't run on them.

It would be nice if it did.

**Comment by fenetikm** [comment]
Pretty sure this will be fixed by the other cache issue #19 . Will close this and we can re-open if needed.
